From 7885cd78c0285698d898f9127a0d56a911b1dc5f Mon Sep 17 00:00:00 2001
From: Oscar Lesta <oscar.lesta@gmail.com>
Date: Sun, 28 Jul 2024 18:05:23 -0300
Subject: [PATCH 1/2] Fix build. Crashes on startup thou.

Closes #1.

Getting a segfault on BWindows::Bounds(), near WindowTab::GetPreferredSize(),
albeit with different parameter types for GCC2 vs x86 (or x86_64).
---
 src/LClickFilter.cpp | 17 ++---------------
 src/LClickFilter.h   | 13 -------------
 src/Makefile         |  4 ++--
 3 files changed, 4 insertions(+), 30 deletions(-)

diff --git a/src/LClickFilter.cpp b/src/LClickFilter.cpp
index 9390e82..ed6429c 100644
--- a/src/LClickFilter.cpp
+++ b/src/LClickFilter.cpp
@@ -5,6 +5,8 @@
 #include "LClickFilter.h"
 #include <View.h>	// for B_PRIMARY_MOUSE_BUTTON, etc...
 
+#define PRINT
+
 const uint32 LClickFilter::kDrag			= 'Drag';
 const uint32 LClickFilter::kDoubleClick		= 'DClk';
 const uint32 LClickFilter::kSingleClick		= 'SClk';
@@ -90,18 +92,3 @@ LClickFilter::Filter( BMessage* message, BHandler** target )
 
 	return B_DISPATCH_MESSAGE;
 }
-
-LMouseTrackingThread::LMouseTrackingThread()
-:	LThread( "LMouseTrackingThread" )
-{
-}
-
-LMouseTrackingThread::~LMouseTrackingThread()
-{
-}
-
-status_t
-LMouseTrackingThread::Main( void )
-{
-	return B_NO_ERROR;
-}
diff --git a/src/LClickFilter.h b/src/LClickFilter.h
index 67f110f..708d484 100644
--- a/src/LClickFilter.h
+++ b/src/LClickFilter.h
@@ -5,7 +5,6 @@
 #define _L_CLICK_FILTER_H
 
 #include <MessageFilter.h>
-#include "LThread.h"
 
 class LClickFilter : public BMessageFilter
 {
@@ -27,16 +26,4 @@ class LClickFilter : public BMessageFilter
 	uint32					mLastMouseButton;
 };
 
-class LMouseTrackingThread : public LThread
-{
-public:
-							LMouseTrackingThread();
-	virtual					~LMouseTrackingThread();
-	virtual status_t		Main( void );
-
-protected:
-
-private:
-};
-
 #endif // _L_CLICK_FILTER_H
diff --git a/src/Makefile b/src/Makefile
index 816ca2b..14444bb 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -5,7 +5,7 @@
 ## architecture of Haiku.
 
 # The name of the binary.
-NAME = LaunchPad
+NAME = ../LaunchPad
 
 # The type of binary, must be one of:
 #	APP:	Application
@@ -64,7 +64,7 @@ RSRCS =
 #	- 	if your library does not follow the standard library naming scheme,
 #		you need to specify the path to the library and it's name.
 #		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
-LIBS = be
+LIBS = be prefs $(STDCPPLIBS)
 
 #	Specify additional paths to directories following the standard libXXX.so
 #	or libXXX.a naming scheme. You can specify full paths or paths relative

From 317dc63ef4336ea1b83d152325494988f90224e0 Mon Sep 17 00:00:00 2001
From: Oscar Lesta <oscar.lesta@gmail.com>
Date: Sun, 28 Jul 2024 18:24:16 -0300
Subject: [PATCH 2/2] Avoid start-up crash. Better fixes welcomed!

---
 src/PaletteWindow.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/PaletteWindow.cpp b/src/PaletteWindow.cpp
index 3202f9b..63f1489 100644
--- a/src/PaletteWindow.cpp
+++ b/src/PaletteWindow.cpp
@@ -179,6 +179,9 @@ WindowTab::Draw( BRect )
 void
 WindowTab::GetPreferredSize( float *w, float *h )
 {
+	if (!Window())
+		return;
+
 	if ( Orientation() == B_HORIZONTAL )
 	{
 		*w = (Window()->Bounds()).Width();
